octodns/provider/mythicbeasts.py

```diff
@@ -0,0 +1,420 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+import re
+
+from requests import Session
+from logging import getLogger
+
+from ..record import Record
+from .base import BaseProvider
+
+
+def add_trailing_dot(value):
+    '''
+    Add trailing dots to values
+    '''
+    assert value, 'Missing value'
+    assert value[-1] != '.', 'Value already has trailing dot'
+    return value + '.'
+
+
+def remove_trailing_dot(value):
+    '''
+    Remove trailing dots from values
+    '''
+    assert value, 'Missing value'
+    assert value[-1] == '.', 'Value already missing trailing dot'
+    return value[:-1]
+
+
+class MythicBeastsProvider(BaseProvider):
+    '''
+    Mythic Beasts DNS API Provider
+
+    mythicbeasts:
+      class: octodns.provider.mythicbeasts.MythicBeastsProvider
+        zones:
+          my-zone: 'password'
+    '''
+
+    SUPPORTS_GEO = False
+    SUPPORTS_DYNAMIC = False
+    SUPPORTS = set(('A', 'AAAA', 'ALIAS', 'CNAME', 'MX', 'NS',
+                    'SRV', 'SSHFP', 'CAA', 'TXT'))
+    BASE = 'https://dnsapi.mythic-beasts.com/'
+
+    def __init__(self, identifier, passwords, *args, **kwargs):
+        self.log = getLogger('MythicBeastsProvider[{}]'.format(identifier))
+
+        assert isinstance(passwords, dict), 'Missing passwords'
+
+        self.log.debug(
+            '__init__: id=%s, registered zones; %s',
+            identifier,
+            passwords.keys())
+        super(MythicBeastsProvider, self).__init__(identifier, *args, **kwargs)
+
+        self._passwords = passwords
+        sess = Session()
+        self._sess = sess
+
+    def _request(self, method, path, data=None):
+        self.log.debug('_request: method=%s, path=%s data=%s',
+                       method, path, data)
+
+        resp = self._sess.request(method, path, data=data)
+        self.log.debug(
+            '_request:   status=%d data=%s',
+            resp.status_code,
+            resp.text[:20])
+
+        if resp.status_code == 401:
+            raise Exception('Mythic Beasts unauthorized for domain: {}'
+                            .format(data['domain']))
+        resp.raise_for_status()
+        return resp
+
+    def _post(self, data=None):
+        return self._request('POST', self.BASE, data=data)
+
+    def records(self, zone):
+        assert zone in self._passwords, 'Missing password for domain: {}' \
+            .format(remove_trailing_dot(zone))
+
+        return self._post({
+            'domain': remove_trailing_dot(zone),
+            'password': self._passwords[zone],
+            'showall': 0,
+            'command': 'LIST',
+        })
+
+    @staticmethod
```

Comment:
Any reason in particular these are `@staticmethod`? Obviously works fine, but it doesn't line up with the way all of the other providers have thing laid out.

Reply:
Originally it wasn't but I may have been a bit over zealous with pylint at some point as that's the only reason that comes to mind (given it has no need for self and should be static but is also fine to be nonstatic.